core/interfaces.go

```diff
@@ -26,6 +26,9 @@ import (
 type Scheduler interface {
 	// Subscribe registers a callback for fetching a duty.
 	Subscribe(func(context.Context, Duty, FetchArgSet) error)
+
+	// GetDuty returns the argSet for a duty if resolved already.
+	GetDuty(ctx context.Context, duty Duty) (FetchArgSet, error)
```

Comment:
can remove parameter names in interface if it isn't confusing (which it isn't in this case), see `Subscribe` above

Reply:
done